Running `butane` tests fail unless async feature is enabled

Most of the `butane/` tests fail if `async` feature isnt enabled with

```
$ cd butane
$ cargo test
...
error[E0046]: not all trait items implemented, missing: `save_many_to_many_async`
   --> butane/tests/basic.rs:104:1
    |
104 | #[model]
    | ^^^^^^^^ missing `save_many_to_many_async` in implementation
    |
    = note: this error originates in the attribute macro `model` (in Nightly builds, run with -Z macro-backtrace for more info)
    = help: implement the missing item: `async fn save_many_to_many_async<impl ConnectionMethodsAsync>(&mut self, _: &impl ConnectionMethodsAsync) -> Result<(), butane::Error> where impl ConnectionMethodsAsync: ConnectionMethodsAsync { todo!() }`

error[E0599]: no method named `save` found for struct `Post` in the current scope
   --> butane/tests/common/blog.rs:190:10
    |
30  | #[model]
    | -------- method `save` not found for this struct
...
190 |     post.save(conn).await.unwrap();
    |          ^^^^ method not found in `Post`
    |
    = help: items from traits can only be used if the trait is in scope
help: the following traits which provide `save` are implemented but not in scope; perhaps you want to import one of them
    |
3   + use butane::DataObjectOpsSync;
    |
3   + use butane::ManyOpsSync;
    |
3   + use butane_core::DataObjectOpsAsync;
    |
3   + use butane_core::many::ManyOpsAsync;
    |
```

The three doctest fails with
```
error[E0046]: not all trait items implemented, missing: `save_many_to_many_async`
  --> butane/src/lib.rs:144:1
   |
12 | #[model]
   | ^^^^^^^^ missing `save_many_to_many_async` in implementation
   |
   = note: this error originates in the attribute macro `model` (in Nightly builds, run with -Z macro-backtrace for more info)
   = help: implement the missing item: `async fn save_many_to_many_async<impl ConnectionMethodsAsync>(&mut self, _: &impl ConnectionMethodsAsync) -> Result<(), butane::Error> where impl ConnectionMethodsAsync: ConnectionMethodsAsync { todo!() }`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0046`.
Couldn't compile the test.

failures:
    butane/src/lib.rs - find (line 135)
```

--- a/butane/Cargo.toml
+++ b/butane/Cargo.toml
@@ -64,8 +64,20 @@ uuid_for_test = { package = "uuid", version = "1.2", features = ["v4"] }
 all-features = true
 
 [[test]]
+name = "basic"
+required-features = ["async"]
+
+[[test]]
+name = "custom_enum_derived"
+required-features = ["async"]
+
+[[test]]
 name = "custom_pg"
 required-features = ["pg"]
+
+[[test]]
+name = "custom_type"
+required-features = ["async"]
 
 [[test]]
 name = "fake"
@@ -73,12 +85,24 @@ required-features = ["fake"]
 
 [[test]]
 name = "json"
-required-features = ["json"]
+required-features = ["async", "json"]
+
+[[test]]
+name = "many"
+required-features = ["async"]
+
+[[test]]
+name = "nullable"
+required-features = ["async"]
 
 [[test]]
 name = "pool"
 required-features = ["r2d2", "deadpool"]
 
 [[test]]
+name = "query"
+required-features = ["async"]
+
+[[test]]
 name = "uuid"
-required-features = ["uuid"]
+required-features = ["async", "uuid"]

--- a/butane/src/lib.rs
+++ b/butane/src/lib.rs
@@ -56,23 +56,28 @@ pub mod db;
 ///    is single literal, it is assumed to be used to match the
 ///    primary key.
 ///
-/// # Examples
-/// ```
-/// # use butane::query::BoolExpr;
-/// # use butane_codegen::model;
-/// # use butane_codegen::filter;
-/// #[model]
-/// struct Contestant {
-///   #[pk]
-///   name: String,
-///   rank: i32,
-///   nationality: String
-/// }
-/// let e: BoolExpr = filter!(Contestant, nationality == "US" && rank < 42);
-/// let first_place = 1;
-/// let e2 = filter!(Contestant, rank == { first_place });
-/// let e3 = filter!(Contestant, name.like("A%"));
-/// ```
+#[cfg_attr(
+    feature = "async",
+    doc = r##"
+# Examples
+```
+# use butane::query::BoolExpr;
+# use butane_codegen::model;
+# use butane_codegen::filter;
+#[model]
+struct Contestant {
+    #[pk]
+    name: String,
+    rank: i32,
+    nationality: String
+}
+let e: BoolExpr = filter!(Contestant, nationality == "US" && rank < 42);
+let first_place = 1;
+let e2 = filter!(Contestant, rank == { first_place });
+let e3 = filter!(Contestant, name.like("A%"));
+```
+"##
+)]
 ///
 /// [`BoolExpr`]: crate::query::BoolExpr
 /// [`Query`]: crate::query::Query
@@ -83,22 +88,27 @@ pub use butane_codegen::filter;
 /// Use as `query!(Foo, expr)`, where `Foo` is a model type. Returns [`Query`]`<Foo>`.
 ///
 /// Shorthand for `Foo::query().filter(`[`filter`]`!(Foo, expr))`
-//
-/// # Examples
-/// ```
-/// # use butane::query::*;
-/// # use butane_codegen::model;
-/// # use butane::query;
-/// # use butane::prelude::*;
-/// #[model]
-/// struct Contestant {
-///   #[pk]
-///   name: String,
-///   rank: i32,
-///   nationality: String
-/// }
-/// let top_tier: Query<Contestant> = query!(Contestant, rank <= 10);
-/// ```
+///
+#[cfg_attr(
+    feature = "async",
+    doc = r##"
+# Examples
+```
+# use butane::query::*;
+# use butane_codegen::model;
+# use butane::query;
+# use butane::prelude::*;
+#[model]
+struct Contestant {
+    #[pk]
+    name: String,
+    rank: i32,
+    nationality: String
+}
+let top_tier: Query<Contestant> = query!(Contestant, rank <= 10);
+```
+"##
+)]
 ///
 /// [`filter]: crate::filter
 /// [`Query`]: crate::query::Query
@@ -131,26 +141,31 @@ macro_rules! colname {
 ///
 /// This macro is for convenience -- it does nothing that can't be done with `query!` or `filter!`.
 ///
-/// # Examples
-/// ```no_run
-/// # use butane::db::ConnectionSpec;
-/// # use butane::query::BoolExpr;
-/// # use butane_codegen::model;
-/// # use butane::prelude::*;
-/// # use butane::query;
-/// # use butane::find;
-/// # use butane::DataObject;
-/// #[model]
-/// struct Contestant {
-///   #[pk]
-///   name: String,
-///   rank: i32,
-///   nationality: String
-/// }
-///
-/// let conn = butane::db::connect(&ConnectionSpec::new("sqlite", "foo.db")).unwrap();
-/// let alice: Result<Contestant, butane::Error> = find!(Contestant, name == "Alice", &conn);
-/// ```
+#[cfg_attr(
+    feature = "async",
+    doc = r##"
+# Examples
+```no_run
+# use butane::db::ConnectionSpec;
+# use butane::query::BoolExpr;
+# use butane_codegen::model;
+# use butane::prelude::*;
+# use butane::query;
+# use butane::find;
+# use butane::DataObject;
+#[model]
+struct Contestant {
+    #[pk]
+    name: String,
+    rank: i32,
+    nationality: String
+}
+
+let conn = butane::db::connect(&ConnectionSpec::new("sqlite", "foo.db")).unwrap();
+let alice: Result<Contestant, butane::Error> = find!(Contestant, name == "Alice", &conn);
+```
+"##
+)]
 ///
 /// [`filter]: crate::filter
 /// [`Result`]: crate::Result

--- a/butane/tests/common/mod.rs
+++ b/butane/tests/common/mod.rs
@@ -1,1 +1,3 @@
+//! Common test utilities.
+#[cfg(feature = "async")]
 pub mod blog;

